wimm.Secundatives.UnitTests/Extensions/AsMaybe_Test.cs

```diff
@@ -44,6 +45,47 @@ public void AsMaybeNullable_Value_ReturnsSome()
             Assert.Equal(value.Value, res.Value);
         }
 
+        [Fact]
+        public async Task AsMaybeTaskClass_Null_ReturnsNone()
+        {
+            var value = Task.FromResult((string)null);
+
+            var res = await value.AsMaybe();
+
+            Assert.Equal(Maybe<string>.None, res);
+        }
+
+
+        [Fact]
+        public async Task AsMaybeTaskClass_Value_ReturnsSome()
+        {
+            var value = Task.FromResult("doot");
+
+            var res = await value.AsMaybe();
+
+            Assert.Equal("doot", res.Value);
+        }
+
+        [Fact]
+        public async Task AsMaybeTaskNullable_Null_ReturnsNone()
+        {
+            var value = Task.FromResult((int?)null);
+
+            var res = await value.AsMaybe();
+
+            Assert.Equal(Maybe<int>.None, res);
+        }
+
+        [Fact]
+        public async Task AsMaybeTaskNullable_Value_ReturnsSome()
+        {
+            var value = Task.FromResult((int?)25);
+
+            var res = await value.AsMaybe();
+
+            Assert.Equal(25, res.Value);
+        }
+
```

Comment:
what happens if you `AsMaybe` a non-nullable type? I assume it just always returns a non-None?

Reply:
You can't. There's no defined method for non-nullable types

```
var i = 100;
i.AsMaybe(); //Compiler error
```

It's intended just as a compatibility layer between the runtime's different ways of saying "This thing might not exist" and the Maybe<T> struct.

Reply:
For Task<T> it's a bit wonk. I'm not huge on it but using it on a Task that returns a non-nullable type doesn't make a ton of sense but since Task<T> is a class it just creates a Maybe<Task<T>> which I'll admit isn't my favourite behaviour but generally it should be really obvious and handled by the type system. 

I could just create a struct one that handles 

Task<T> where T : struct and then check for nullability but I think that we'd have to re-jig all of the AsMaybe's to act the same way and tbh I don't think that .AsMaybe() should be just used as a way to avoid the Maybe<T> ctor.